Zend/zend_compile.c

```diff
@@ -3533,99 +3559,107 @@ uint32_t zend_compile_args(
 			arg_count++;
 		}
 
-		/* Treat passing of $GLOBALS the same as passing a call.
-		 * This will error at runtime if the argument is by-ref. */
-		if (zend_is_call(arg) || is_globals_fetch(arg)) {
-			zend_compile_var(&arg_node, arg, BP_VAR_R, 0);
-			if (arg_node.op_type & (IS_CONST|IS_TMP_VAR)) {
-				/* Function call was converted into builtin instruction */
-				if (!fbc || ARG_MUST_BE_SENT_BY_REF(fbc, arg_num)) {
-					opcode = ZEND_SEND_VAL_EX;
-				} else {
-					opcode = ZEND_SEND_VAL;
-				}
-			} else {
-				if (fbc && arg_num != (uint32_t) -1) {
-					if (ARG_MUST_BE_SENT_BY_REF(fbc, arg_num)) {
-						opcode = ZEND_SEND_VAR_NO_REF;
-					} else if (ARG_MAY_BE_SENT_BY_REF(fbc, arg_num)) {
-						opcode = ZEND_SEND_VAL;
+		if (arg && arg->kind != ZEND_AST_PARTIAL_ARG) {
```

Comment:
Can you treat `arg->kind == ZEND_AST_PARTIAL_ARG` here early, put the rest into else if and thus save one level of indent?